MediaWikiKit/MediaWikiKit/MWKArticle.m

```diff
@@ -570,9 +570,21 @@ - (NSAttributedString*)summaryHTML {
         if (!nonEmptyParagraphsWithSelectedChildren.count) {
             continue;
         }
-        NSString* lineSeparatedParagraphs = [nonEmptyParagraphsWithSelectedChildren componentsJoinedByString:@"<br/><br/>"];
-        NSData* xpathData                 = [lineSeparatedParagraphs dataUsingEncoding:NSUTF8StringEncoding];
-        return [[NSAttributedString alloc] initWithHTMLData:xpathData site:self.site];
+        NSString* lineSeparatedParagraphs = [nonEmptyParagraphsWithSelectedChildren componentsJoinedByString:@"</p><p>"];
+        lineSeparatedParagraphs = [NSString stringWithFormat:@"<p>%@</p>", lineSeparatedParagraphs];
+        NSData* xpathData = [lineSeparatedParagraphs dataUsingEncoding:NSUTF8StringEncoding];
+
+        NSAttributedString* returnValue = [[NSAttributedString alloc] initWithHTMLData:xpathData site:self.site];
+
+        // Remove trailing line break. Prob move this to a category.
+        if (returnValue.length) {
```

Comment:
there's already a method for this: `stringByTrimmingCharactersInSet:(NSCharacterSet*)` which returns:
> A new string made by removing from both ends of the receiver characters contained in set. If the receiver is composed entirely of characters from set, the empty string is returned.